querydsl-sql/src/main/java/com/querydsl/sql/types/JSR310InstantType.java

```diff
@@ -35,12 +34,20 @@ public Class<Instant> getReturnedClass() {
     @Nullable
     @Override
     public Instant getValue(ResultSet rs, int startIndex) throws SQLException {
-        Timestamp timestamp = rs.getTimestamp(startIndex, utc());
-        return timestamp != null ? timestamp.toInstant() : null;
+        try {
+            return rs.getObject(startIndex, Instant.class);
+        } catch (SQLException e) {
+            Timestamp timestamp = rs.getTimestamp(startIndex);
```

Comment:
Either we do need the test cases or we can get rid of the catch clauses as well.

Reply:
In the end, I mostly reverted this change since, as it turns out, few JDBC drivers support Instant natively. So continuing to use Timestamp's conversion to/from Instant seems like the best choice.